File: project_kanban_to_form/static/src/js/project.js
/*
    © 2017 Savoir-faire Linux <https://savoirfairelinux.com>
    © 2022 - today Numigi <https://numigi.com>
    License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL.html).
*/
odoo.define('project_kanban_to_form', function (require) {
'use strict';

require('project.update_kanban');
var KanbanRecord = require('web.KanbanRecord');

KanbanRecord.include({
    _openRecord() {
        if (this.modelName === 'project.project') {
            this._openProjectForm()
        } else {
            this._super.apply(this, arguments);
        }
    },
    /**
     * This method was copied from the method _openRecord of web.KanbanRecord
     */
    _openProjectForm() {
        if (this.$el.hasClass('o_currently_dragged')) {
            return;
        }
        var editMode = this.$el.hasClass('oe_kanban_global_click_edit');
        this.trigger_up('open_record', {
            id: this.db_id,
            mode: editMode ? 'edit' : 'readonly',
        });
    }
});
});

File: project_archive_improved/models/project.py
# -*- coding: utf-8 -*-
# © 2017 Savoir-faire Linux
# License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL).

from odoo import api, models


class Project(models.Model):

    _inherit = 'project.project'

    @api.multi
    def write(self, vals):
        super(Project, self).write(vals)
        if 'active' in vals:
            self.mapped('analytic_account_id').write({
                'active': vals['active'],
            })

        return True
